JL.Core/Mining/MiningUtils.cs

```diff
@@ -440,6 +561,76 @@ public static async Task MineToFile(LookupResult lookupResult, string currentTex
         Utils.Logger.Information("Mined {PrimarySpelling}", lookupResult.PrimarySpelling);
     }
 
+    public static async ValueTask<bool[]?> CheckDuplicates(List<LookupResult> lookupResults, string currentText, int currentCharPosition)
+    {
+        List<Note> notes = [];
+        List<int> positions = [];
+        bool[] results = new bool[lookupResults.Count];
+        for (int i = 0; i < lookupResults.Count; i++)
+        {
+            LookupResult lookupResult = lookupResults[i];
+
+            Dictionary<MineType, AnkiConfig>? ankiConfigDict = await AnkiConfig.ReadAnkiConfig().ConfigureAwait(false);
+            if (ankiConfigDict is null)
+            {
+                continue;
+            }
+
+            AnkiConfig? ankiConfig;
+            if (DictUtils.s_wordDictTypes.Contains(lookupResult.Dict.Type))
+            {
+                ankiConfig = ankiConfigDict.GetValueOrDefault(MineType.Word);
+            }
+            else if (DictUtils.s_kanjiDictTypes.Contains(lookupResult.Dict.Type))
+            {
+                ankiConfig = ankiConfigDict.GetValueOrDefault(MineType.Kanji);
+            }
+            else if (DictUtils.s_nameDictTypes.Contains(lookupResult.Dict.Type))
+            {
+                ankiConfig = ankiConfigDict.GetValueOrDefault(MineType.Name);
+            }
+            else
+            {
+                ankiConfig = ankiConfigDict.GetValueOrDefault(MineType.Other);
+            }
+
+            if (ankiConfig is null)
+            {
+                continue;
+            }
+
+            Dictionary<string, JLField> userFields = ankiConfig.Fields;
+            JLField firstValue = userFields.First().Value;
+            string? value = GetMiningParameter(firstValue, lookupResult, currentText, currentCharPosition, true);
+            if (value is null)
+            {
+                continue;
+            }
+
+            Dictionary<string, string> fields = ConvertFields(userFields, new()
+            {
+                [firstValue] = value
+            });
```

Comment:
```suggestion
            (string firstFieldName, JLField firstField) = userFields.First();
            string? firstFieldValue = GetMiningParameter(firstField, lookupResult, currentText, currentCharPosition);
            if (value is null)
            {
                continue;
            }

            Dictionary<string, string> fields = new(1, StringComparer.Ordinal)
            {
                {  firstFieldName, "firstFieldValue" }
            };
```

Reply:
Ah yeah this makes a lot more sense... My limited C# knowledge really showing here.

Reply:
Technically speaking, calling `First()` on `userFields` isn't guaranteed to return the first field, since a `Dictionary` does not guarantee any order. However, as an implementation detail, `Dictionary` does maintain insertion order unless there are many items causing collisions, which may lead to the dictionary being rebuilt and losing the insertion order. I will properly address this after migrating JL to .NET 9 by using an `OrderedDictionary` instead of a `Dictionary` for the `Fields` property.